### ISSUE TITLE
Visibility based on radio buttons

### ISSUE THREAD

**Issue by hashitha** [WPF-Forge/Forge.Forms#28]
In the following case, is there any way we can make the textview visible only if we select the first option 

```
  <select
    name="CalculateDates"
    defaultValue="LMP"
    as="radiobuttons"
    label="Select">
    <option
      value="LMP date:">LMP</option>
    <option
      value="EDD date:">EDD</option>
  </select>


<textarea  
        type="string"
        name="Comments"
        label="EDD comments"></textarea>
```

**Comment by hashitha** [comment]
maybe we can base the visibility on the selected index?

**Comment by edongashi** [comment]
`textarea isvisible="{Binding CalculateDates|???}"`

I have to think about the converter part. It's unfortunate that binding expressions dont support converter arguments. Maybe we should add that to the grammar.

Something like
`{Binding CalculateDates|IsEqual('LMP')}`

**Comment by hashitha** [comment]
I think `{Binding CalculateDates|IsEqual('LMP')}` will work well

**Comment by edongashi** [comment]
Anywhere that you have a bool you can create custom expressions like that one, so this already works. We only need to implement converter parameters for better reusability.

**Comment by edongashi** [comment]
Converters can now have a parameter using the following syntax:

```
|IsEqual('String')
|IsEqual(3)
|IsEqual(25.3)
|IsEqual(true)
|IsEqual(false)
|IsEqual(null)
```

The `IsEqual` is just a demonstration. You can create custom factories and register them in `Resource.ValueConverterFactories`. You cannot have a dynamic value in converters.

 **Note** This is not `Binding.ConverterParameter` that is passed to `IValueConverter.Convert`. It was too messy and too late to implement it. Instead, the parameter is parsed from the expression and is passed to corresponding factory in the new property: `Resource.ValueConverterFactories`, which creates a new converter with the parameter value baked in.

For backwards compatibility we now have two dictionaries for storing converters globally:

`Resource.ValueConverters` is a dictionary of `string : IValueConverter` and `Resource.ValueConverterFactories` is a dictionary of `string : func(object -> IValueConverter)`.

Feel free to register custom converters at application startup.

**Comment by hashitha** [comment]
I have tested this and it works well on Textview and datetime but I can't get this to work on `<text>` field

**Comment by edongashi** [comment]
`text visible="false"` does not seem to work. I will see what's wrong

**Comment by hashitha** [comment]
It is the same with Heading tag

**Comment by edongashi** [comment]
fixed in 42e23e238e9c909baa8d6a48f10cff9897dc9daa for all non-input elements

**Comment by hashitha** [comment]
This is working well now
